docker/go-control-plane/controlplane/callbacks.go

```diff
@@ -0,0 +1,163 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package controlplane provides components for the common control plane used
+// for testing
+package controlplane
+
+import (
+	"context"
+	"log"
+	"os"
+	"sync"
+
+	pb_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	pb_discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/v3"
+)
+
+// Implementation of the server.Callbacks interface that implements the behavior
+// required by our tests.
```

Comment:
The `server/v3` package provides a convenience type `CallbackFuncs` which can be used instead of trying to implement the `Callbacks` interface. With the `CallbackFuncs` approach, you fill in fields in a struct that you care about, instead of having to implement all methods in the interface (most of which you don't care about).